src/client/game/board.ts

```diff
@@ -165,13 +165,26 @@ export class Board {
     if (p.session_id === this.localPlayerID) {
       this.team = team;
     }
+    // Update join buttons to reflect new state
+    this.updateJoinButtons();
 
     this.createTile(p, team, force);
     this.updateGameStatus();
     this.updateInteraction();
     toggleEndTurnButton(this.currentTurn, this.team);
   }
 
+  eject(playerID: string) {
+    removeTile(playerID);
+    if (this.spymasters.team1 === playerID) {
```

Comment:
Previously we were just calling `removeTile()` here. But what if a spymaster is leaving? We need to update the relevant join buttons to allow others to join the team as a spymaster if this is the case, hence this addition.